frontend/src/components/projectListView/ProjectListView.tsx

```diff
@@ -0,0 +1,252 @@
+
+  const [documentationDescription, setDocumentationDescription] =
+    useState<string>('');
+  const [ongoingDescription, setOngoingDescription] = useState<string>('');
+  const [todoDescription, setTodoDescription] = useState<string>('');
+  const [doneDescription, setDoneDescription] = useState<string>('');
+
+  const [documentationEditing, setDocumentationEditing] =
+    useState<boolean>(false);
+  const [ongoingEditing, setOngoingEditing] = useState<boolean>(false);
+  const [todoEditing, setTodoEditing] = useState<boolean>(false);
+  const [doneEditing, setDoneEditing] = useState<boolean>(false);
+
+  const { id } = useParams();
+  const [userId, setUserId] = useState<string>('');
+
+  useEffect(() => {
+    const filterTaskByStatus = (status: string) => {
+      return allTasks.filter((task) => task.status === status);
+    };
+    const setUser = () => {
+      const auth = getAuth();
+      onAuthStateChanged(auth, async (user: User | null) => {
+        if (user) {
+          setUserId(user.uid);
+        }
+      });
+    };
+    setDocumentation(filterTaskByStatus('documentation'));
+    setOngoing(filterTaskByStatus('ongoing'));
+    setTodo(filterTaskByStatus('to_do'));
+    setDone(filterTaskByStatus('done'));
+    setUser();
+  }, [allTasks]);
+
+  const addNewTask = (
+    setTasks: React.Dispatch<React.SetStateAction<Task[]>>,
+    description: string,
+    tasks: Task[],
+    status: string
+  ) => {
+    if (id && description.trim()) {
+      setTasks([
+        ...tasks,
+        {
+          id: uuid(), // for mocked data
+          description,
+          status,
+          due_date: '',
+          assignee: '',
+          completed: false,
+          priority: '',
+          project_id: +id,
+          user_id: userId
+        }
+      ]);
+    }
+  };
+
+  const onDragEnd = (result: any) => {
+    const { source, destination } = result;
+    if (!destination) return;
+    if (source.droppableId !== destination.droppableId) {
+      setAllTasks(
+        allTasks.map((task) =>
+          task.id === +result.draggableId
+            ? { ...task, status: destination.droppableId }
+            : task
+        )
+      );
+    } else {
+      const changeOrder = (
+        setTasks: React.Dispatch<React.SetStateAction<Task[]>>,
+        tasksArr: Task[]
+      ) => {
+        setTasks((tasksArr) => {
+          const tasks = [...tasksArr];
+          const [removed] = tasks.splice(source.index, 1);
+          tasks.splice(destination.index, 0, removed);
+          return tasks;
+        });
+      };
+      if (destination.droppableId === 'documentation') {
+        changeOrder(setDocumentation, documentation);
+      } else if (destination.droppableId === 'ongoing') {
+        changeOrder(setOngoing, ongoing);
+      } else if (destination.droppableId === 'to_do') {
+        changeOrder(setTodo, todo);
+      } else if (destination.droppableId === 'done') {
+        changeOrder(setDone, done);
+      }
```

Comment:
Use a switch case here instead. Also create constants for these different `droppableId`s
Something like:
`
const droppableIdDocumentation = 'documentation'
const droppableIdOngoing = 'ongoing'
so on...
`